api/src/main/java/org/apache/iceberg/Transaction.java

```diff
@@ -162,4 +162,7 @@ default UpdateStatistics updateStatistics() {
    * @throws CommitFailedException If the updates cannot be committed due to conflicts.
    */
   void commitTransaction();
+
+  /** Rollbacks the commits created by compaction in case it causes conflict with the updates. */
+  void rollbackCompactionOnConflicts();
```

Comment:
is there a way we can determine we can rollback a snapshot without introducing a new terminology of compaction? Currently we use `RewriteFiles` which means it's always a `replace` type operation. Isn't that sufficient condition to trigger auto-rollback on conflict?

Reply:
ACK : will start off with this being table level property.